File: src/main/java/com/fractalwrench/bpp/Logger.java
package com.fractalwrench.bpp;

public final class Logger {

    private static boolean enabled = true;

    private Logger() {
    }

    public static void setEnabled(boolean enabled) {
        Logger.enabled = enabled;
    }

    public static void log(String msg) {
        if (enabled) {
            System.out.println(msg);
        }
    }

    public static void log(String msg, Throwable throwable) {
        if (enabled) {
            System.out.println(msg);
            throwable.printStackTrace();
        }
    }
}

File: src/main/java/com/fractalwrench/bpp/Logger.kt
package com.fractalwrench.bpp

object Logger {

    private var enabled = true

    fun setEnabled(enabled: Boolean) {
        Logger.enabled = enabled
    }

    fun log(msg: String) {
        if (enabled) {
            println(msg)
        }
    }

    fun log(msg: String, throwable: Throwable) {
        if (enabled) {
            println(msg)
            throwable.printStackTrace()
        }
    }
}
